CodingGuidelines.md

```diff
@@ -0,0 +1,26 @@
+# Coding Guidelines
+
+## C# General
+
+In order to improve team wide readablity and lower onboarding time we agreed to stick to defined coding guidelines. These coding guidelines help us writing better and more coherent code, both on an individual level, as well as across individual developers since the style becomes more aligned.
+
+The coding guidelines can be found [here](https://csharpcodingguidelines.com/).
+Feel free to download the .PDF for later references.
+
+## IDE
+
+At Triple we are not tied to a single operating system, and so it happens that also among the .NET team there is cross-platform development. In addition to the free choice of IDE, there are quite a few flavors possible. However, in order to reduce startup and onboarding time, we should strive to facilitate cross-platform development as best as possible. This includes the generation of a README file, with instructions for environment specific setups. At the least, it should contain the instructions for the cross-platform editor  [Visual Studio Code](https://code.visualstudio.com/).
+
+## Testing
+
+We thorougly appreciate it when business logic is well covered by a test suite. This helps with extending functionality, refactoring existing code, and general readability.
+In order to promote productivity, we should use the [NUnit](https://nunit.org/) testing framework, in combination with the mocking framework [Moq](https://github.com/Moq) by default.
+
+### Further reading resources
+
+* [Clean Code series](https://www.pearson.com/us/higher-education/series/Robert-C-Martin-Series/348084.html) by Robert C. Martin
```

Comment:
The clean series is dated, and wasn't that great when it was recent.
I recommend not recommending it.

Reply:
I've gone through the [article](https://qntm.org/clean) you suggested why not to recommend _Clean Clode_. It has been 2 years now since I read the series and looking back at the few examples and arguments the author provides I have to admit that some of them are fuzzy indeed and the author has a point. 

Concerning the 'bad parts' in the book, I quote:
_"Is the whole book like this? Pretty much, yeah. Clean Code mixes together a disarming combination of strong, timeless advice and advice which is highly questionable or dated or both."_

I suggest focusing on the _'disarming combination of strong timeless advice'_. Let's not forget the author has had decades of hands-on experience in our field, which we should not casually dismiss. Let's instead read carefully and discuss to come to common ground, and use this to follow-up.

In the end, the author of the article found a resource worth recommending: [A Philosophy of Software Design (2018)](https://www.goodreads.com/en/book/show/39996759-a-philosophy-of-software-design). Have you read this? Accordingly, _"it focuses more on the practice of software design at a higher level than it does on the writing or critiquing of actual individual lines of code"_, so it would also be good to find a good resource on that. 
I will look into the book in order to make a good comparison.
In the comments we can find more examples of books recommended by readers. And seen the amount of them, there is no concensus on what to read, and what not. 

Therefore I think providing a reading list of resources we've enjoyed to suggest further reading is fine, as it opens discussion and we might learn a thing or two. Would you add your recommendations?